ibis/expr/operations/reductions.py

```diff
@@ -106,7 +106,7 @@ def output_dtype(self):
 class Mean(Filterable, Reduction):
     arg = rlz.column(rlz.numeric)
 
-    @immutable_property
+    @initialized
```

Comment:
Perhaps we should just call it `initialize`?

Reply:
`intialized` is fine for now IMO